Library/Homebrew/cmd/update.rb

```diff
@@ -228,6 +219,16 @@ def pull!(options = {})
       @initial_branch = ""
     end
 
+    unless `git status --untracked-files=all --porcelain 2>/dev/null`.chomp.empty?
+      if ARGV.verbose?
+        puts "Stashing uncommitted changes to #{repository}."
+        system "git", "status", "--short", "--untracked-files=all"
+      end
+      safe_system "git", "stash", "save", "--include-untracked", *@quiet_args
+      safe_system "git", "reset", "--hard"
```

Comment:
Just to clarify, it does so for repositories with local modifications. Otherwise this line obviously wouldn't be executed.

Also, isn't this line a no-op and can be omitted as the immediately preceding `git stash save` already returns the working directory to a clean state? Or am I overlooking a case where that's not true?